### ISSUE TITLE
Useless library. Unfortunately

### ISSUE THREAD

**Issue by denvist** [meyfa/php-svg#30]
Code from example:

`<?php`
`require_once __DIR__.'/php-svg/autoloader.php';`
`use SVG\SVGImage;`
`$image = SVGImage::fromFile("bodygraph.svg");`
`$rasterImage = $image->toRasterImage(600, 600);`
`header('Content-Type: image/png');`
`imagepng($rasterImage);`
`?>`

Original SVG Image: ![bodygraph.svg](https://user-images.githubusercontent.com/7146527/30519698-53ec7a52-9ba5-11e7-8536-9664198f2489.png)

Rendered PNG Image: ![rendered image](https://user-images.githubusercontent.com/7146527/30519699-53ef7054-9ba5-11e7-8c5b-c72f2631d32b.png)

**Comment by denvist** [comment]
Even i modify code to render SVG Image:
`<?php`
`require_once __DIR__.'/php-svg/autoloader.php';`
`use SVG\SVGImage;`
`$image = SVGImage::fromFile("bodygraph.svg");`
`header('Content-Type: image/svg+xml');`
`echo $image;`

Result is not good: where is no gradient and notes.
![Rendered SVG Image](https://user-images.githubusercontent.com/7146527/30519732-271f5dae-9ba6-11e7-912f-8644e23670ee.png)

**Comment by meyfa** [comment]
I agree, that is a terrible result. Thank you for making the effort of reporting your experience.

Could you perhaps send a link to the original SVG file so that I have something to test with?

**Comment by denvist** [comment]
[Link](http://xn--d1abaamcctt0b3a2a8a.xn--p1ai/swe/bodygraph.svg) to image.

**Comment by meyfa** [comment]
There is some progress. The result is still nowhere near the original, but it's getting better.

Task list for tracking things that need to be implemented:

- [X] MoveTo commands with additional coordinate pairs need to be treated as implicit LineTo commands (e5ae072)
- [X] correct opacity calculation (1518365)
- [ ] gradients
- [ ] translate transforms
- [ ] matrix transforms
- [ ] fill rules (tracked by #28)

![r1](https://user-images.githubusercontent.com/3982806/30782275-18a23a92-a12f-11e7-9c4b-2abaa2bf913a.png)

**Comment by ningappa** [comment]
any update on this?

**Comment by jan-potuznik** [comment]
I tried Your own example provided in README.md under "Rasterizing" and it ignores stroke-width.
